src/new.tsx

```diff
@@ -93,16 +99,86 @@ function Charming() {
     }
   }, [data, points]);
 
+  React.useEffect(() => {
+    // Data is required to determine grapheme cluster breaks in pasted strings
+    if (!data) return;
+
+    const onKeyDown = (e: KeyboardEvent) => {
+      if (e.key == "/") {
+        openSearch();
+        e.preventDefault();
```

Comment:
in general, it’s safer to preventDefault at the start of a block that needs it, in case we throw an exception